eglib/src/gstr.c

```diff
@@ -251,10 +258,16 @@ g_strerror (gint errnum)
 			error_messages [errnum] = g_strdup (buff);
 		if (buff != tmp_buff)
 			g_free (buff);
-#else
+#else /* HAVE_STRERROR_R_XSI */
+		buff = strerror_r (errnum, buff, buff_len);
```

Comment:
If we are to trust the docs, this is ok, as buff would alternatively point to an immutable buffer.